### PR TITLE
Minor fixes

### PR DESCRIPTION
**Pull request by pszal**
- addresses my first comment from #374 (i.e., deterministic server selection)
- removes `PathSegment.segment_id` #385 
- fixes CPS bug with syncing local master after down-path is registered
  <a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/netsec-ethz/scion/pull/402%23issuecomment-145022165%22%2C%20%22https%3A//github.com/netsec-ethz/scion/pull/402%23issuecomment-145029875%22%2C%20%22https%3A//github.com/netsec-ethz/scion/pull/402%23issuecomment-145035022%22%2C%20%22https%3A//github.com/netsec-ethz/scion/commit/0af863c9e1837e469dbfbec6edc345487ed9fd4a%23commitcomment-13587467%22%2C%20%22https%3A//github.com/netsec-ethz/scion/commit/0af863c9e1837e469dbfbec6edc345487ed9fd4a%23commitcomment-13587666%22%2C%20%22https%3A//github.com/netsec-ethz/scion/commit/0af863c9e1837e469dbfbec6edc345487ed9fd4a%23commitcomment-13588213%22%2C%20%22https%3A//github.com/netsec-ethz/scion/commit/0af863c9e1837e469dbfbec6edc345487ed9fd4a%23commitcomment-13588236%22%2C%20%22https%3A//github.com/netsec-ethz/scion/commit/0af863c9e1837e469dbfbec6edc345487ed9fd4a%23commitcomment-13588252%22%2C%20%22https%3A//github.com/netsec-ethz/scion/pull/402%23discussion_r41405217%22%2C%20%22https%3A//github.com/netsec-ethz/scion/pull/402%23discussion_r41405806%22%2C%20%22https%3A//github.com/netsec-ethz/scion/pull/402%23discussion_r41405871%22%2C%20%22https%3A//github.com/netsec-ethz/scion/pull/402%23issuecomment-146236884%22%2C%20%22https%3A//github.com/netsec-ethz/scion/pull/402%23discussion_r41499023%22%5D%2C%20%22comments%22%3A%20%7B%22General%20Comment%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/netsec-ethz/scion/pull/402%23issuecomment-145022165%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22%40shitz%20you%20may%20want%20to%20look%20at%20this.%22%2C%20%22created_at%22%3A%20%222015-10-02T13%3A36%3A24Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/8159928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/pszalach%22%7D%7D%2C%20%7B%22body%22%3A%20%22If%20we%20want%20to%20forward%20a%20request%20of%20the%20same%20source%20_always_%20to%20the%20same%20path-/cert-%20etc%20server%20then%20LGTM.%22%2C%20%22created_at%22%3A%20%222015-10-02T14%3A01%3A17Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/3840858%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/shitz%22%7D%7D%2C%20%7B%22body%22%3A%20%22merging%20%28%40kormat%20can%20review%20after%20he%20is%20back%29%22%2C%20%22created_at%22%3A%20%222015-10-02T14%3A20%3A48Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/8159928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/pszalach%22%7D%7D%2C%20%7B%22body%22%3A%20%22One%20minor%20comment%2C%20LGTM%20otherwise%20%3A%29%22%2C%20%22created_at%22%3A%20%222015-10-07T15%3A39%3A25Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6919822%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/kormat%22%7D%7D%5D%2C%20%22title%22%3A%20%22General%20Comment%22%7D%2C%20%22Commit%200af863c9e1837e469dbfbec6edc345487ed9fd4a%20endhost/sciond.py%205%20404%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/netsec-ethz/scion/commit/0af863c9e1837e469dbfbec6edc345487ed9fd4a%23commitcomment-13587467%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22%40pszalach%20This%20should%20be%20%60segment.get_hops_hash%28%29%60%2C%20right%3F%22%2C%20%22created_at%22%3A%20%222015-10-05T09%3A19%3A39Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6560106%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/mskd96%22%7D%7D%2C%20%7B%22body%22%3A%20%22This%20should%20indeed%20be%20%60get_hops_hash%28%29%60.%20%22%2C%20%22created_at%22%3A%20%222015-10-05T09%3A29%3A12Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/3840858%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/shitz%22%7D%7D%2C%20%7B%22body%22%3A%20%22No%2C%20it%20is%20fine.%20In%20%60PathSegmentDBRecord%60%27s%20%60__init__%60%20I%20do%20%60self.id%20%3D%20pcb.get_hops_hash%28%29%60%22%2C%20%22created_at%22%3A%20%222015-10-05T09%3A55%3A18Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/8159928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/pszalach%22%7D%7D%2C%20%7B%22body%22%3A%20%22oh%2C%20now%20I%20see%20that%20%60__call__%60%20returns%20PCBs%2C%20yes%2C%20should%20be%20as%20%40mskd96%20said%22%2C%20%22created_at%22%3A%20%222015-10-05T09%3A56%3A43Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/8159928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/pszalach%22%7D%7D%2C%20%7B%22body%22%3A%20%22I%27ll%20create%20a%20PR%20soon%22%2C%20%22created_at%22%3A%20%222015-10-05T09%3A57%3A35Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/8159928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/pszalach%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20endhost/sciond.py%3AL404%20%280af863c%29%22%7D%2C%20%22Pull%20ef73ef73b637b37f662f5b7ec1c8c904182aec21%20infrastructure/beacon_server.py%2022%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/netsec-ethz/scion/pull/402%23discussion_r41405217%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22It%20would%20be%20much%20better%20to%20use%20%60pcb.short_desc%60%20here.%22%2C%20%22created_at%22%3A%20%222015-10-07T15%3A33%3A06Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6919822%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/kormat%22%7D%7D%2C%20%7B%22body%22%3A%20%22fixed%20with%20%23407%5Cn%5CnOn%207%20October%202015%20at%2017%3A33%2C%20Stephen%20Shirley%20%3Cnotifications%40github.com%3E%5Cnwrote%3A%5Cn%5Cn%3E%20In%20infrastructure/beacon_server.py%5Cn%3E%20%3Chttps%3A//github.com/netsec-ethz/scion/pull/402%23discussion_r41405217%3E%3A%5Cn%3E%5Cn%3E%20%3E%20%40%40%20-1411%2C7%20%2B1408%2C7%20%40%40%20def%20register_up_segments%28self%29%3A%5Cn%3E%20%3E%20%20%20%20%20%20%20%20%20%20%20%20%20%20except%20SCIONServiceLookupError%20as%20e%3A%5Cn%3E%20%3E%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20logging.warning%28%5C%22Unable%20to%20send%20up%20path%20registration%3A%20%25s%5C%22%2C%20e%29%5Cn%3E%20%3E%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20continue%5Cn%3E%20%3E%20-%20%20%20%20%20%20%20%20%20%20%20%20logging.info%28%5C%22Up%20path%20registered%3A%20%25s%5C%22%2C%20pcb.segment_id%29%5Cn%3E%20%3E%20%2B%20%20%20%20%20%20%20%20%20%20%20%20logging.info%28%5C%22Up%20path%20registered%3A%20%25s%5C%22%2C%20pcb.get_hops_hash%28%29%29%5Cn%3E%5Cn%3E%20It%20would%20be%20much%20better%20to%20use%20pcb.short_desc%20here.%5Cn%3E%5Cn%3E%20%5Cu2014%5Cn%3E%20Reply%20to%20this%20email%20directly%20or%20view%20it%20on%20GitHub%5Cn%3E%20%3Chttps%3A//github.com/netsec-ethz/scion/pull/402/files%23r41405217%3E.%5Cn%3E%5Cn%22%2C%20%22created_at%22%3A%20%222015-10-08T10%3A42%3A11Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/8159928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/pszalach%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20infrastructure/beacon_server.py%3AL1408-1415%22%7D%2C%20%22Pull%20ef73ef73b637b37f662f5b7ec1c8c904182aec21%20lib/packet/host_addr.py%206%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/netsec-ethz/scion/pull/402%23discussion_r41405806%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22What%20is%20this%20for%3F%22%2C%20%22created_at%22%3A%20%222015-10-07T15%3A37%3A39Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6919822%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/kormat%22%7D%7D%2C%20%7B%22body%22%3A%20%22Ohh.%20To%20allow%20sorting.%20Ok.%5Cr%5Cn%3A%2B1%3A%20%22%2C%20%22created_at%22%3A%20%222015-10-07T15%3A38%3A08Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6919822%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/kormat%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20lib/packet/host_addr.py%3AL102-111%22%7D%7D%7D'></a>
  <a href='https://www.codereviewhub.com/'></a>
- [ ] <a href='#crh-comment-General Comment'></a> &nbsp;<b><a href='https://github.com/netsec-ethz/scion/pull/402#issuecomment-145022165'>General Comment</a></b>
- <a href='https://github.com/pszalach'></a> @shitz you may want to look at this.
- <a href='https://github.com/shitz'></a> If we want to forward a request of the same source _always_ to the same path-/cert- etc server then LGTM.
- <a href='https://github.com/pszalach'></a> merging (@kormat can review after he is back)
- <a href='https://github.com/kormat'></a> One minor comment, LGTM otherwise :)
- [ ] <a href='#crh-comment-Commit 0af863c9e1837e469dbfbec6edc345487ed9fd4a endhost/sciond.py 5 404'></a> &nbsp;<b><a href='https://github.com/netsec-ethz/scion/commit/0af863c9e1837e469dbfbec6edc345487ed9fd4a#commitcomment-13587467'>File: endhost/sciond.py:L404 (0af863c)</a></b>
- <a href='https://github.com/mskd96'></a> @pszalach This should be `segment.get_hops_hash()`, right?
- <a href='https://github.com/shitz'></a> This should indeed be `get_hops_hash()`.
- <a href='https://github.com/pszalach'></a> No, it is fine. In `PathSegmentDBRecord`'s `__init__` I do `self.id = pcb.get_hops_hash()`
- <a href='https://github.com/pszalach'></a> oh, now I see that `__call__` returns PCBs, yes, should be as @mskd96 said
- <a href='https://github.com/pszalach'></a> I'll create a PR soon
- [ ] <a href='#crh-comment-Pull ef73ef73b637b37f662f5b7ec1c8c904182aec21 infrastructure/beacon_server.py 22'></a> &nbsp;<b><a href='https://github.com/netsec-ethz/scion/pull/402#discussion_r41405217'>File: infrastructure/beacon_server.py:L1408-1415</a></b>
- <a href='https://github.com/kormat'></a> It would be much better to use `pcb.short_desc` here.
- <a href='https://github.com/pszalach'></a> fixed with #407
  On 7 October 2015 at 17:33, Stephen Shirley notifications@github.com
  wrote:
  > In infrastructure/beacon_server.py
  > https://github.com/netsec-ethz/scion/pull/402#discussion_r41405217:
  >
  > > @@ -1411,7 +1408,7 @@ def register_up_segments(self):
  > >              except SCIONServiceLookupError as e:
  > >                  logging.warning("Unable to send up path registration: %s", e)
  > >                  continue
  > > -            logging.info("Up path registered: %s", pcb.segment_id)
  > > +            logging.info("Up path registered: %s", pcb.get_hops_hash())
  >
  > It would be much better to use pcb.short_desc here.
  >
  > —
  > Reply to this email directly or view it on GitHub
  > https://github.com/netsec-ethz/scion/pull/402/files#r41405217.
  >
- [x] <a href='#crh-comment-Pull ef73ef73b637b37f662f5b7ec1c8c904182aec21 lib/packet/host_addr.py 6'></a> &nbsp;<b><a href='https://github.com/netsec-ethz/scion/pull/402#discussion_r41405806'>File: lib/packet/host_addr.py:L102-111</a></b>
- <a href='https://github.com/kormat'></a> What is this for?
- <a href='https://github.com/kormat'></a> Ohh. To allow sorting. Ok.
  :+1:

<a href='https://www.codereviewhub.com/netsec-ethz/scion/pull/402?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/netsec-ethz/scion/pull/402?approve=1'></a>&nbsp;<a href='https://github.com/netsec-ethz/scion/pull/402'></a>
<a href='#crh-end'></a>

### CODE REVIEW

File: infrastructure/beacon_server.py [hunk 1]
```diff
@@ -1411,7 +1408,7 @@ def register_up_segments(self):
             except SCIONServiceLookupError as e:
                 logging.warning("Unable to send up path registration: %s", e)
                 continue
-            logging.info("Up path registered: %s", pcb.segment_id)
+            logging.info("Up path registered: %s", pcb.get_hops_hash())
```
Comment (kormat) [inline infrastructure/beacon_server.py:1411]: It would be much better to use `pcb.short_desc` here.

File: lib/packet/host_addr.py [hunk 2]
```diff
@@ -102,6 +102,9 @@ def __len__(self):
     def __eq__(self, other):
         return (self.TYPE == other.TYPE) and (self.addr == other.addr)
 
+    def __lt__(self, other):
+        return str(self) < str(other)
+
```
Comment (kormat) [inline lib/packet/host_addr.py:107]: What is this for?

Comment (kormat) [inline lib/packet/host_addr.py:107]: Ohh. To allow sorting. Ok.
:+1:

File: infrastructure/beacon_server.py (hunk 1 above)
Comment (pszal) [inline infrastructure/beacon_server.py:1411]: fixed with #407

**Pull request merged**